File: Easy/Java/463. Island Perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int island = 0;

        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == 1){
                    island += 4;
                    if(0 < i && grid[i - 1][j] == 1)
                        island -= 2;
                    if(0 < j && grid[i][j - 1] == 1)
                        island -= 2;
                }
            }
        }

        return island;
    }
}

File: Easy/Python/463. Island Perimeter.py
class Solution(object):
    def islandPerimeter(self, grid):
        island = 0

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    island += 4
                    if 0 < i and grid[i - 1][j] == 1:
                        island -= 2
                    if 0 < j and grid[i][j - 1] == 1:
                        island -= 2

        return island
